### PR TITLE
owls85464 - add MII dynamic update tests for changing work manager config

### PR DESCRIPTION
**Pull request by alai8**
This is for merging to `dynamicupdate` branch

Added new integration test `ItMiiDynamicUpdate` under integration-tests/src/test/java/oracle/weblogic/kubernetes/ for testing dynamic update of work manager configurations. First update adds a new work manager, min and max threads constraints, and a second update changes the count of the min and max threads constraint.

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3120/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+                domainUid,
+                domainNamespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(domainExists(domainUid, DOMAIN_VERSION, domainNamespace));
+  }
+
+  @AfterAll
+  public void tearDownAll() {
+    deletePersistentVolume(pvName);
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java:185]: we don't need this, the test framework will cleanup all pvs

Comment (alai8) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java]: Thanks. I have removed it.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java [hunk 2]
```diff
@@ -1510,6 +1510,30 @@ public static String patchDomainResourceWithNewRestartVersion(
...
   }
 
+  /**
+   * Patch the domain resource with a new model configMap.
+   *
+   * @param domainResourceName name of the domain resource
+   * @param namespace Kubernetes namespace that the domain is hosted
+   * @param configMapName name of the configMap to be set in spec.configuration.model.configMap
+   */
+  public static void patchDomainResourceWithModelConfigMap(
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java:1520]: can you add a method in Domain.java and call that method from this method, no implementation in TestActions class.

**Review by vanajamukkara — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+            "weblogicenc", domainNamespace),
+             String.format("createSecret failed for %s", encryptionSecretName));
+
+    logger.info("Create database secret");
+    final String dbSecretName = domainUid  + "-db-secret";
+    assertDoesNotThrow(() -> createDatabaseSecret(dbSecretName, "scott",
+            "tiger", "jdbc:oracle:thin:localhost:/ORCLCDB", domainNamespace),
+             String.format("createSecret failed for %s", dbSecretName));
+
+    createConfigMapAndVerify(
```
[the remark this answers is not in the document]
Comment (alai8) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java]: I have update the test to create a config map without any files.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+    }
+  }
+
+  /**
+   * Check server logs are written on PV.
+   */
+  @Test
+  @Order(1)
+  @DisplayName("Check the server logs are written on PV, look for string RUNNING in server log")
+  public void testServerLogsAreOnPV() {
```
[the remark this answers is not in the document]
Comment (alai8) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java]: Removed testServerLogsAreOnPV(). Thanks

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+   * section of the domain resource.
+   * The configmap has multiple sparse WDT model files that define 
+   * a JDBCSystemResource, a JMSSystemResource and a WLDFSystemResource.
+   * Verify all the SystemResource configurations using the rest API call 
+   * using the public nodeport of the administration server.
+   */
+  @Test
+  @Order(2)
+  @DisplayName("Verify the pre-configured SystemResources in a model-in-image domain")
+  public void testMiiCheckSystemResources() {
```
[the remark this answers is not in the document]
Comment (alai8) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDynamicUpdate.java]: Removed testMiiCheckSystemResources()

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java (hunk 2 above)
Comment (alai8) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java:1495]: Sure. I have also move logic for patchDomainResourceWithModelConfigMap to Domain and call the methods from TestActions. Also did the same to patchDomainResourceWithNewRestartVersion.

**Review by vanajamukkara — APPROVED** [review]
LG

**Pull request merged**